android-lib/src/androidTest/java/com/kinvey/androidTest/store/data/DataStoreMultiInsertTestKotlin.kt

```diff
@@ -0,0 +1,1954 @@
+import com.kinvey.androidTest.TestManager
+import com.kinvey.androidTest.model.EntitySet
+import com.kinvey.androidTest.model.Person
+import com.kinvey.androidTest.network.MockMultiInsertNetworkManager
+import com.kinvey.java.AbstractClient
+import com.kinvey.java.Constants
+import com.kinvey.java.KinveySaveBatchException
+import com.kinvey.java.Logger
+import com.kinvey.java.Query
+import com.kinvey.java.core.AbstractKinveyClient
+import com.kinvey.java.core.KinveyClientCallback
+import com.kinvey.java.core.KinveyJsonResponseException
+import com.kinvey.java.model.KinveyBatchInsertError
+import com.kinvey.java.model.KinveyPullResponse
+import com.kinvey.java.model.KinveyReadResponse
+import com.kinvey.java.store.StoreType
+import com.kinvey.java.sync.dto.SyncItem
+import com.kinvey.java.sync.dto.SyncRequest
+
+import org.junit.Before
+import org.junit.Ignore
+import org.junit.Test
+import org.junit.runner.RunWith
+
+import java.io.IOException
+import java.lang.reflect.Field
+import java.util.ArrayList
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertNull
+import org.junit.Assert.assertTrue
+
+@RunWith(AndroidJUnit4::class)
+@SmallTest
+@Ignore("Ignored tests in Kotlin for now while Kotlin migration is not done")
+class DataStoreMultiInsertTestKotlin {
+
+    private var client: Client<User>? = null
+
+    @Before
+    @Throws(InterruptedException::class, IOException::class)
+    fun setUp() {
+        val mMockContext = InstrumentationRegistry.getInstrumentation().targetContext
+        client = Client.Builder<User>(mMockContext).build()
+        client?.enableDebugLogging()
+        AbstractClient.KINVEY_API_VERSION = "5"
+        val latch = CountDownLatch(1)
+        var looperThread: LooperThread? = null
+        if (client?.isUserLoggedIn == false) {
+            looperThread = LooperThread(Runnable {
+                try {
+                    UserStore.login(TestManager.USERNAME, TestManager.PASSWORD, client as Client<User>, object : KinveyClientCallback<User> {
+                        override fun onSuccess(result: User) {
+                            assertNotNull(result)
+                            latch.countDown()
+                        }
+
+                        override fun onFailure(error: Throwable) {
+                            assertNull(error)
+                            latch.countDown()
+                        }
+                    })
+                } catch (e: IOException) {
+                    e.printStackTrace()
+                }
+            })
+            looperThread.start()
+        } else {
+            latch.countDown()
+        }
+        latch.await()
+        looperThread?.mHandler?.sendMessage(Message())
+    }
+
+    private class DefaultKinveyClientCallback<T : GenericJson> (private val latch: CountDownLatch) : KinveyClientCallback<T> {
```

Comment:
is possible to create static classes in Kotlin?

Reply:
If we need this to prevent memory-leak then in `Kotlin` if we just create an `inner` class by default that it's not keeps link to `outer` class and we can use it safely. So we can keep changes as it is.